Sources/Sentry/SentryTraceState.m

```diff
@@ -0,0 +1,117 @@
+#import "SentryTraceState.h"
+#import "SentryDsn.h"
+#import "SentryLog.h"
+#import "SentryOptions+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySerialization.h"
+#import "SentryTracer.h"
+#import "SentryUser.h"
+
+@implementation SentryTraceStateUser
+
+- (instancetype)initWithUserId:(NSString *)userId segment:(NSString *)segment
+{
+    if (self = [super init]) {
+        _userId = userId;
+        _segment = segment;
+    }
+    return self;
+}
+
+- (instancetype)initWithUser:(nullable SentryUser *)user
+{
+    return [self initWithUserId:user.userId segment:[user.data objectForKey:@"segment"]];
+}
+
+@end
+
+@implementation SentryTraceState
+
+- (instancetype)initWithTraceId:(SentryId *)traceId
+                      publicKey:(NSString *)publicKey
+                    releaseName:(nullable NSString *)releaseName
+                    environment:(nullable NSString *)environment
+                    transaction:(nullable NSString *)transaction
+                           user:(nullable SentryTraceStateUser *)user
+{
+    if (self = [super init]) {
+        _traceId = traceId;
+        _publicKey = publicKey;
+        _environment = environment;
+        _releaseName = releaseName;
+        _transaction = transaction;
+        _user = user;
+    }
+    return self;
+}
+
+- (nullable instancetype)initWithScope:(SentryScope *)scope options:(SentryOptions *)options
+{
+    if (![scope.span isKindOfClass:[SentryTracer class]])
+        return nil;
+    return [self initWithTracer:scope.span scope:scope options:options];
+}
+
+- (instancetype)initWithTracer:(SentryTracer *)tracer
+                         scope:(nullable SentryScope *)scope
+                       options:(SentryOptions *)options
+{
+
+    return [self initWithTraceId:tracer.context.traceId
+                       publicKey:options.parsedDsn.url.user
+                     releaseName:options.releaseName
+                     environment:options.environment
+                     transaction:tracer.name
+                            user:[[SentryTraceStateUser alloc] initWithUser:scope.userObject]];
+}
+
+- (NSString *)toHTTPHeader
+{
+    NSError *error;
+    NSDictionary *json = [self serialize];
+    NSData *data = [SentrySerialization dataWithJSONObject:json error:&error];
+
+    if (nil != error) {
+        [SentryLog
+            logWithMessage:[NSString stringWithFormat:@"Couldn't encode trace state: %@", error]
+                  andLevel:kSentryLevelError];
+        return nil;
+    }
+
+    NSString *encodedData =
+        [[data base64EncodedStringWithOptions:0] stringByReplacingOccurrencesOfString:@"="
```

Comment:
`m`: I think it's only about trailing `=` and not all of them, see [docs](https://develop.sentry.dev/sdk/performance/trace-context/#tracestate-headers)

> Strip trailing padding characters (=), since this is a reserved character.

I think you are missing this step in Java, @marandaneto.

Reply:
You are totally right, but base64 encode only uses '=' as trailing pads, so I choose the simplest function to replace characters in a string.

Reply:
Ah, sorry, my bad. Of course, base64 only uses `=` as padding, so we are safe. I add a comment for this.